writer.go

```diff
@@ -401,24 +431,31 @@ func (p *MediaPlaylist) InsertSegment(seqNo uint64, seg *MediaSegment) error {
 // next chunk. Secondly it appends one chunk to the tail of chunk slice. Useful for sliding playlists.
 // This operation does reset cache.
 func (p *MediaPlaylist) Slide(uri string, duration float64, title string) {
+	p.lock.Lock()
+	defer p.lock.Unlock()
 	if p.Live && p.count >= p.winsize {
-		p.Remove()
+		p.remove()
 	} else if p.Live {
 		p.SeqNo++
 	}
-	p.Append(uri, duration, title)
+	p.append(uri, duration, title)
 }
 
 // Reset playlist cache. Next called Encode() will regenerate playlist from the chunk slice.
 func (p *MediaPlaylist) ResetCache() {
-	p.buf.Reset()
+	p.lock.Lock()
+	p.dirty = true
+	p.lock.Unlock()
```

Comment:
For code consistency, please p.lock.Lock() followed by defer p.lock.Unlock()

Reply:
fixed